includes/class-wc-product-variable.php

```diff
@@ -449,47 +449,31 @@ public function validate_props() {
 	}
 
 	/**
-	 * Save data (either create or update depending on if we are working on an existing product).
+	 * Do any extra processing needed before the actual product save
+	 * (but after triggering the 'woocommerce_before_..._object_save' action)
 	 *
-	 * @since 3.0.0
+	 * @return mixed A state value that will be passed to after_data_store_save_or_update.
 	 */
-	public function save() {
-		$this->validate_props();
-
-		if ( ! $this->data_store ) {
-			return $this->get_id();
-		}
-
-		/**
-		 * Trigger action before saving to the DB. Allows you to adjust object props before save.
-		 *
-		 * @param WC_Data          $this The object being saved.
-		 * @param WC_Data_Store_WP $data_store The data store persisting the data.
-		 */
-		do_action( 'woocommerce_before_' . $this->object_type . '_object_save', $this, $this->data_store );
-
+	protected function before_data_store_save_or_update() {
```

Comment:
If we remove the `save` method, wouldn't that break backwards compatibility?

Reply:
The method exists already in the base class (`abstract-wc-product.php`). It can be removed here, thus removing some code suplication, thanks to the new `before_data_store_save_or_update` and `after_data_store_save_or_update` methods